### DISCUSSION TITLE
Unable to Save Plots to File

### DISCUSSION THREAD

**Discussion by DannBoMambo** [gnudatalanguage/gdl#1219] [Q&A]
Hi,

I've been running into this error: `% Warning: failed to create temporary PostScript file` whenever I attempt to save a plot to a file, either through Device or cgPlot. It does generate an incomplete postscript file (i.e. missing graphics like cgText, cgColorfill, cgLegend etc) in the active directory. The command line window crashes shortly after.

Even something as simple as:
`set_plot,'ps'`
`device,filename='test.ps',xsize=16,ysize=16,/color,decomposed=0`
`x=findgen(30)`
`plot,x,x*x`
`device,/close`
returns this error.

For a bit of background, I was a frequent IDL user a few years ago and recently have setup GDL on my windows machine. I have the coyote and astrolib libraries installed in C:/Program Files/gnudatalanguage/share/gnudatalanguage/lib.

Is there some kind of environment variable that sets the path of where it would try to save temporary files? Is Windows blocking this action somehow? 

On a related note, back when I used IDL on a Linux machine I could set the env. vars using .bashrc, but I haven't had much luck figuring out how to do that with Windows so that it's automatically done every login.

**Comment by slayoo** [comment]
I'm not a windows user, but Stack Overflow seems to have a page on the .bashrc equivalent for Windows - perhaps this will help:
https://stackoverflow.com/questions/588321/can-system-environment-variables-be-set-via-windows-logon-scripts

**Comment by GillesDuvert** [comment]
hello @DannBoMambo I can reproduce the error on a windows 10 (virtual machine on my linux). The file 'test.ps' is however in the current directory, and is good. And the command window does not crash shortly after (as far as I can test).
It could be that the 'temporary postscript file' is a special behaviour of the plplot library under windows.
Difficult to go further,could  somebody with a better knowledge of plplot-on-windows or Windows investitgate?
